arch/arm/core/cortex_m/mpu/arm_mpu_v8_internal.h

```diff
@@ -73,6 +73,16 @@ static inline void _get_mpu_ram_region_attr(arm_mpu_region_attr_t *p_attr,
 	p_attr->r_limit = REGION_LIMIT_ADDR(base, size);
 }
 
+#ifdef CONFIG_RAM_FUNCTION
+static inline void _get_mpu_ramfunc_region_attr(arm_mpu_region_attr_t *p_attr,
```

Comment:
Just to confirm with you, @qianfan-Zhao 
This attribute-setting shall be identical to _get_mpu_ram_region_attr(....), except for _not_ setting the XN bit, correct?

Reply:
BTW, this code will _not_ compile in; if there is no USERSPACE || MPU_STACK_GUARD || APPLICATION_MEMORY
Maybe you need to implement this function before line 59

Reply:
> This attribute-setting shall be identical to _get_mpu_ram_region_attr(....), except for not setting the XN bit, correct?

Yes

Reply:
This is still not good; we do not want all the rest of the implementation to be compiled in. See comment above.